src/test/java/org/folio/rest/api/HridSettingsStorageTest.java

```diff
@@ -0,0 +1,473 @@
+  }
+
+  @Test
+  public void cannotUpdateHridSettingsID()
+      throws InterruptedException, ExecutionException, TimeoutException {
+    log.info("Starting cannotUpdateHridSettingsID()");
+
+    final CompletableFuture<Response> originalGetCompleted = new CompletableFuture<>();
+
+    client.get(InterfaceUrls.hridSettingsStorageUrl(""), TENANT_ID, json(originalGetCompleted));
+
+    final Response originalGetResponse = originalGetCompleted.get(5, SECONDS);
+
+    assertThat(originalGetResponse.getStatusCode(), is(200));
+
+    final HridSettings originalHridSettings =
+        originalGetResponse.getJson().mapTo(HridSettings.class);
+
+    final String uuid = UUID.randomUUID().toString();
+
+    final HridSettings newHridSettings = new HridSettings()
+        .withId(uuid)
+        .withInstance(new HridSetting().withPrefix("inst").withStartNumber(100))
+        .withHolding(new HridSetting().withPrefix("hold").withStartNumber(200))
+        .withItem(new HridSetting().withPrefix("item").withStartNumber(500));
+
+    final CompletableFuture<Response> putCompleted = new CompletableFuture<>();
+
+    client.put(InterfaceUrls.hridSettingsStorageUrl(""), newHridSettings, TENANT_ID,
+        empty(putCompleted));
+
+    final Response putResponse = putCompleted.get(5, SECONDS);
+
+    assertThat(putResponse.getStatusCode(), is(204));
+
+    final CompletableFuture<Response> getAfterUpdateCompleted = new CompletableFuture<>();
+
+    client.get(InterfaceUrls.hridSettingsStorageUrl(""), TENANT_ID, json(getAfterUpdateCompleted));
+
+    final Response getAfterUpdateResponse = getAfterUpdateCompleted.get(5, SECONDS);
+
+    final HridSettings actualHridSettings =
+        getAfterUpdateResponse.getJson().mapTo(HridSettings.class);
+
+    assertThat(actualHridSettings.getId(), not(uuid));
+    assertThat(actualHridSettings.getId(), is(originalHridSettings.getId()));
+
+    assertThat(actualHridSettings.getInstance(), is(notNullValue()));
+    assertThat(actualHridSettings.getInstance().getPrefix(),
+        is(newHridSettings.getInstance().getPrefix()));
+    assertThat(actualHridSettings.getInstance().getStartNumber(),
+        is(newHridSettings.getInstance().getStartNumber()));
+
+    assertThat(actualHridSettings.getHolding(), is(notNullValue()));
+    assertThat(actualHridSettings.getHolding().getPrefix(),
+        is(newHridSettings.getHolding().getPrefix()));
+    assertThat(actualHridSettings.getHolding().getStartNumber(),
+        is(newHridSettings.getHolding().getStartNumber()));
+
+    assertThat(actualHridSettings.getItem(), is(notNullValue()));
+    assertThat(actualHridSettings.getItem().getPrefix(),
+        is(newHridSettings.getItem().getPrefix()));
+    assertThat(actualHridSettings.getItem().getStartNumber(),
+        is(newHridSettings.getItem().getStartNumber()));
+
+    log.info("Finished cannotUpdateHridSettingsID()");
+  }
+
+  @Test
+  public void canGetNextInstanceHrid(TestContext testContext)
```

Comment:
This test appears not to use the API, is that the case?

If so, is there a more appropriate place for it? Does it rely on setup that is done by test base?

Reply:
Many of these tests do not use the API directly. I went back and forth on whether or not to split them out into a separate class but I do, as far as I know, need to extend `TestBase` to leverage the embedded PostgreSQL DB support. I suppose I could have mocked the PostgresClient or duped the DB management code from `StorageTestSuit`. I was trying to save some time, not the best reason for sure.

I'm not sure what a more appropriate place would be. Do you have any suggestions? I see `org.folio.rest.unit`, but these really aren't unit tests.

Reply:
> Many of these tests do not use the API directly. I went back and forth on whether or not to split them out into a separate class but I do, as far as I know, need to extend `TestBase` to leverage the embedded PostgreSQL DB support. I suppose I could have mocked the PostgresClient or duped the DB management code from `StorageTestSuit`. I was trying to save some time, not the best reason for sure.

Yeah, I wondered if this was because infrastructure was needed.

> I'm not sure what a more appropriate place would be. Do you have any suggestions? I see `org.folio.rest.unit`, but these really aren't unit tests.

I believe this is an unsolved issue for the storage modules. The nature of them means that most require infrastructure, making them integration (depending on the use of the term) tests.

We could introduce a `integration` package name. A concern with this (and the `unit` package, is that it can interfere with code coverage reporting in some cases).

Maybe this a technical debt that we address separately, by separating out the database infrastructure preparation and the API preparation and having different base test classes for API and integration (with external resources) tests.

Reply:
These tests _may_ become obsolete once the inventory APIs are integrated with the new HRID management. I needed a way to prevent regression while working on this code.

I like the idea of a tech debt story for test restructuring. Also, maybe move to JUnit5, consider getting rid of the test suite, etc.

Reply:
I'd like to move to JUnit 5, it offers much better options for sharing setup than the suite workaround.

When I last checked, it didn't provide a capability for only running expensive setup once for a set of tests, only sharing the definition. If a pattern has emerged for that, it would help.

Reply:
Due to how much of the behaviour of storage modules are defined by RAML Module Builder, pure unit testing for storage modules is a rather large trade off which doesn't really confirm the actual behaviour

Reply:
I'm comfortable with these tests being a place holder for API tests that demonstrate allocation of HRIDs for new records. Maybe worth a comment noting that is the purpose of them?

Whether we also want more focused tests can be a separate topic to address at that point.